clang/lib/Driver/ToolChains/HIPAMD.cpp

```diff
@@ -78,8 +78,12 @@ void AMDGCN::Linker::constructLldCommand(Compilation &C, const JobAction &JA,
                                          const llvm::opt::ArgList &Args) const {
   // Construct lld command.
   // The output from ld.lld is an HSA code object file.
-  ArgStringList LldArgs{"-flavor", "gnu", "--no-undefined", "-shared",
-                        "-plugin-opt=-amdgpu-internalize-symbols"};
+  ArgStringList LldArgs{"-flavor",
+                        "gnu",
+                        "--no-undefined",
+                        "-shared",
+                        "-plugin-opt=-amdgpu-internalize-symbols",
+                        "-plugin-opt=-sycl-enable-local-accessor"};
```

Comment:
The addition of `-sycl-enable-local-accessor` is added specifically for the device compilation for the clang compilation.  This is being added here for all `lld` usage.  Is there no need to scrutinize here?

Reply:
At this point we are dealing with the device specific command, moreover `AMDGCN` specific device, so the check is not required.